### PR TITLE
CB-1487 Configure Knox SSO to use CDP CP as IdP

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java [hunk 1]
```diff
@@ -897,4 +897,22 @@ private SaltConnector createSaltConnector(GatewayConfig gatewayConfig) {
...
+    @Override
+    public void uploadIdpMetadataXml(List<GatewayConfig> allGatewayConfigs, ExitCriteriaModel exitModel, String metadataXml)
+            throws CloudbreakOrchestratorFailedException {
+        GatewayConfig gateway = getPrimaryGatewayConfig(allGatewayConfigs);
+        Set<String> gatewayTargets = getGatewayPrivateIps(allGatewayConfigs);
+        try (SaltConnector saltConnector = createSaltConnector(gateway)) {
+            if (!gatewayTargets.isEmpty()) {
+                LOGGER.debug("Upload metadata.xml to gateways");
+                byte[] metadataXmlBytes = metadataXml.getBytes();
+                String samlPath = "/var/lib/knox/cloudbreak_resources/";
```
Comment (lajosrodek) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java]: Silly question: How will this get moved to `/etc/knox/sso/saml/metadata.xml` (as expected by the SSO topology)? Is not that something to be done in `orchestrator-salt/src/main/resources/salt/salt/gateway/cm.sls`?

Comment (daszabo) [reply orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java]: Good catch! This is a mistake from my part. I modified this part after testing the feature, but forgot to sync it with the path in knoxsso.xml

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java [hunk 2]
```diff
@@ -530,4 +536,21 @@ private UserManagementBlockingStub newStub(String requestId) {
...
+     */
+    public String getIdentityProviderMetadataXml(String requestId, String accountId) {
+        checkNotNull(accountId);
+
+        UserManagementProto.GetIdPMetadataForWorkloadSSORequest request =
+                UserManagementProto.GetIdPMetadataForWorkloadSSORequest.newBuilder()
+                        .setAccountId(accountId)
+                        .build();
+        UserManagementProto.GetIdPMetadataForWorkloadSSOResponse response = newStub(requestId).getIdPMetadataForWorkloadSSO(request);
+        return response.getMetadata();
```
Comment (keyki) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java]: we should check if the response is null

**Comment by daszabo** [comment]
I will reopen this PR after a small redesign

**Pull request closed without merging**